Client1/Client.py

```diff
@@ -0,0 +1,164 @@
+
+    @classmethod
+    def display_main_menu(cls):
+        response = requests.get("http://127.0.0.1:5000/")
+        menu_options = response.json()
+        print(menu_options)
+        player_choice = input()
+        if player_choice == '1':
+            cls.display_available_rooms()
+        elif player_choice == '2':
+            cls.display_room_details_by_room_name()
+        elif player_choice == '3':
+            cls.host_a_room()
+        elif player_choice == '4':
+            cls.join_a_room()
+        elif player_choice == '5':
+            cls.leave_a_room()
+
+    @classmethod
+    def display_available_rooms(cls):
+        rooms = cls.get_available_rooms()
+        if not rooms:
+            print("There are no available rooms\n")
+        else:
+            print("These are the available game rooms:\n")
+            for room in rooms:
+                print(room)
+        cls.display_main_menu()
+
+    @classmethod
+    def get_available_rooms(cls):
+        url = "http://127.0.0.1:5000/rooms"
+        headers = {'x-access-token': cls.player_token}
+        response = requests.get(url, headers=headers)
+        rooms_as_dict = response.json()
+        rooms_as_obj = [cls.room_dict_to_obj(room_as_dict) for room_as_dict in rooms_as_dict.values()]
+        return rooms_as_obj
+
+    @classmethod
+    def display_room_details_by_room_name(cls):  # Need to check if room exists
+        room_name = input("Please insert the name of the room: ")
+        print("")
+        url = "http://127.0.0.1:5000/rooms/" + room_name
+        headers = {'x-access-token': cls.player_token}
+        response = requests.get(url, headers=headers)
+        room_as_dict = response.json()
+        if not room_as_dict:
+            print(RoomsAPIErrors.ROOM_NAME_DOESNT_EXIST.value)
+        else:
+            room_as_obj = cls.room_dict_to_obj(room_as_dict)
+            print(room_as_obj)
+        cls.display_main_menu()
+
+    @classmethod
+    def host_a_room(cls):
+        room_name = input("Please insert the name of the room: ")
+        print("")
+        data = {'room_name': room_name}
+        url = "http://127.0.0.1:5000/rooms"
+        headers = {'x-access-token': cls.player_token}
+        response = requests.post(url, headers=headers, json=data)
+        print(f"{response.text}\n")
+        if response.text == RoomsAPIErrors.ROOM_NAME_OCCUPIED.value:
+            cls.host_a_room()
+        else:
+            cls.display_main_menu()
+
+    @classmethod
+    def join_a_room(cls):
+        room_name = input("Please insert the name of the room you want to join to: ")
+        url = "http://127.0.0.1:5000/rooms/" + room_name
+        headers = {'x-access-token': cls.player_token}
+        response = requests.put(url, headers=headers)
+        print(f"{response.text}\n")
+        cls.display_main_menu()
+
+    @classmethod
+    def leave_a_room(cls):
+        response = requests.put("http://127.0.0.1:5000/leave_a_room")
```

Comment:
same as join, recommend to "POST"

Reply:
Anther option might be 
POST room/{room_name}/leave. 
but because a player can be only in one room at the same time, your implementation is perfect, and reflect the really better!

Reply:
Anther modeling is that join and leave are related to a user, and not to a room so
POST /user/{user_id}/join_room 
    data : {room_name: "..."}
POST /user/{user_id}/leave_room


But I can't say which option is better. just wanted to expose you to anther way